pkg/dialer/dialer_windows.go

```diff
@@ -17,27 +17,29 @@
 package dialer
 
 import (
+	"context"
 	"net"
 	"os"
-	"syscall"
 	"time"
 
 	winio "github.com/Microsoft/go-winio"
 )
 
 func isNoent(err error) bool {
-	if err != nil {
-		if oerr, ok := err.(*os.PathError); ok {
-			if oerr.Err == syscall.ENOENT {
-				return true
-			}
-		}
-	}
-	return false
+	return os.IsNotExist(err)
 }
 
 func dialer(address string, timeout time.Duration) (net.Conn, error) {
-	return winio.DialPipe(address, &timeout)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
This is already handled by `timeoutDialer` you should be able to just call `winio.DialPipe(address, &timeout)` right?

Reply:
Right. I've completely forgot that `timeoutDialer` handles that.